host-controller/src/main/java/org/jboss/as/host/controller/discovery/DiscoveryOptionsResource.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.host.controller.discovery;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.host.controller.logging.HostControllerLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+
+/**
+ *
+ * @author Alexey Loubyansky
+ */
+public class DiscoveryOptionsResource implements Resource {
+
+    private final Resource delegate;
+
+    public DiscoveryOptionsResource() {
+        this(Resource.Factory.create());
+    }
+
+    DiscoveryOptionsResource(Resource delegate) {
+        if (delegate == null) {
+            throw HostControllerLogger.ROOT_LOGGER.nullVar("delegate");
+        }
```

Comment:
No need to change this; just an FYI.

We put in quite a few of these early in AS7 days, but they can be just an assert delegate != null; If this is null it's a programming error, not a mistake in user input.